File: ICPC-2nd-Year/1st Contest/Robot.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    int T, n, p = 0, n1;
    string In, In2;
    cin >> T;
    int Ans[T];
    for (int i = 0; i < T; i++)
    {
        p = 0;
        cin >> n;
        int L[n] = {0};
        for (int j = 0; j < n; j++)
        {
            cin >> In;
            if (In == "LEFT")
            {
                p--;
                L[j] = -1;
            }
            if (In == "RIGHT")
            {
                p++;
                L[j] = 1;
            }
            if (In == "SAME")
            {
                cin >> In >> n1;
                if (L[n1 - 1] == 1)
                {
                    p++;
                    L[j] = 1;
                }
                if (L[n1 - 1] == -1)
                {
                    p--;
                    L[j] = -1;
                }
            }
        }

        Ans[i] = p;
    }

    for (int i = 0; i < T; i++)
    {
        cout << Ans[i] << endl;
    }
    return 0;
}

File: Contests/CSEC/CSEC Contests -1st-Year/5th Contest/Solutions/Robot.py
x = int(input())

Ans = []

for i in range(x):
    Answer = 0
    Moves = []
    n = int(input())
    for j in range(n):
        ac = input()
        if ac == "LEFT":
            Answer -= 1
            Moves.append(-1)
        elif ac == "RIGHT":
            Answer += 1
            Moves.append(1)
        for i in range(len(Moves)):
            y = "SAME AS {0}".format(i+1)
            if ac == y:
                if Moves[i] == -1:
                    Answer -= 1
                    Moves.append(-1)
                else:
                    Answer += 1
                    Moves.append(1)
    Ans.append(Answer)

for u in Ans:
    print(u)